modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/exp/RexImpTable.java

```diff
@@ -4359,6 +4364,26 @@ private static class DefaultImplementor extends AbstractRexCallImplementor {
     }
   }
 
+  private static class IgniteMethodNameImplementor extends AbstractRexCallImplementor {
```

Comment:
why do we need new implementor?

Reply:
Because `defineMethod` one expects that all methods are defined on calcite's `SqlFunctions`. (1)
And it is not possible to define a method with multiple overloads in `IgniteSqlFunctions`.  (2) 

The second one can be fixed if we to allow defining functions by name w/o specifying their signatures e.g. ROUND function would look like:

> ROUND("round");

This would also required adding methodName to `IgniteSqlFunctions`

Reply:
i did my best ))) to align RexImpTable with calcite and one more change (

Reply:
> Because defineMethod expects that all methods to be defined on calcite's SqlFunctions

that's not true, you could supply reference to particular `java.lang.reflect.Method`. In that case you can use static method from any class

> And it is not possible to define a method with multiple overloads in IgniteSqlFunctions

why do you can define multiple overloads in `SqlFunctions`, but not in `IgniteSqlFunctions`?